fs/squashfs/block.c

```diff
@@ -167,6 +167,8 @@ int squashfs_read_data(struct super_block *sb, u64 index, int length,
 	}
 
 	if (compressed) {
+		if (!msblk->stream)
+			goto read_failure;
```

Comment:
And this from https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=d512584780d3e6a7cacb2f482834849453d444a1